### PR TITLE
[Backup] `az backup restore restore-azurefileshare`: Add `--target-rg-name` parameter to specify the resource group of the destination storage account

### PR DESCRIPTION
**Pull request by zubairabid**
**Related command**

`az backup restore restore-azurefileshare`

**Description**

Resolving work item <https://msazure.visualstudio.com/DefaultCollection/One/_workitems/edit/24475508/>. Allows users to AFS restore to a storage account that is in a different resource group from the vault, without having to provide the full ARM ID of the Storage Account.

**Testing Guide**

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Backup

File: src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_afs_commands.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 class BackupTests(ScenarioTest, unittest.TestCase):
     #@record_only()
     @ResourceGroupPreparer(location="eastus2euap", random_name_length=20)
-    @VaultPreparer()
+    @VaultPreparer(soft_delete=False)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_afs_commands.py:28]: Could you please specify the context of why adding the `soft_delete=False`?

Comment (zubairabid) [reply src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_afs_commands.py:28]: Soft delete is enabled by default, and when running tests, it blocks vault deletion and resource group cleanup as any deleted resources are still listed as protected under it.

Comment (zubairabid) [reply src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_afs_commands.py:28]: It's also present in other tests: https://github.com/Azure/azure-cli/blob/ef2ecdf1c75620fcc278f2d732c5f9b86dc4a7e7/src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_backup_commands.py#L359

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**